### ISSUE TITLE
Discussion: Make CLI flags global

### ISSUE THREAD

**Issue by bartlomieju** [denoland/deno#2228]
With current implementation CLI flags are required after `deno` and before any subcommand:
```shellsession
// this works
$ deno -D info script.ts
// this doesn't work
$ deno info -D script.ts
```

Current solution is not super ergonomic and became apparent during work on #2215 which would require a lot of changes to integration tests in `tests/` - they often go like this:
```
// tests/016_double_await.ts
args: --allow-read --reload tests/016_double_await.ts
output: tests/016_double_await.ts.out

// keeping current solution we'd need
args: --allow-read --reload run tests/016_double_await.ts
output: tests/016_double_await.ts.out
```
With current approach each test has to be visited and `run` subcommand should be added in proper place. Making flags global we could leave tests as they are and only change bits in scripts that run tests.

**Comment by ry** [comment]
`--allow-read` and `--reload` etc should be flags of the run subcoommand, not to deno itself. So you'd only see them when you do `deno help run`.

**Comment by bartlomieju** [comment]
@ry valid point, then `--log-debug` and `--reload` should probably be global anyway. I'd imagine doing `deno fmt --reload script.ts` to update prettier and for `-D`... well, sometimes you just need to check what's going on in any of the commands.

**Comment by bartlomieju** [comment]
Resolved in #2215
